float32 value
float32 lock
---
